TLM/TLM/Custom/PathFinding/CustomPathFind2.cs

```diff
@@ -2818,11 +2818,24 @@ private enum LaneChangingCostCalculationMode {
 					uturnExplored = true;
 				}
 
-				// allow vehicles to ignore strict lane routing when moving off
-				bool relaxedLaneRouting =
+                // allow vehicles to ignore strict lane routing when moving off
+                // disable lane change at the edge of the map.
+                bool disableLaneChangeAtEdge = false;
+                if (Options.disableLaneChangeAtEdge)
+                {
+                    bool isEdgeOfTheMap = false;
+                    isEdgeOfTheMap |= Math.Abs(Singleton<NetManager>.instance.m_segments.m_buffer[nextSegmentId].m_middlePosition.x) > 8000;
```

Comment:
Are these calculations being done every pathfind? If so, the calculated values should be cached to improve performance.

Reply:
More specifically, can we somehow tag the lane or segment as "being on edge of map"?

Is there not some flag (or something) on those segments at the edge? The vanilla game adds (at least in map editor, and possibly also in-game) those green arrow things - check code that places those?

Reply:
Good idea, worth checking

Reply:
The check should be implemented in RoutingManager, not in PathFind. RoutingManager calculates lane transitions once on every network update / settings change.